### ISSUE TITLE
Problem setting blockbogons interface param

### ISSUE THREAD

**Issue by riorescue** [jaredhendrickson13/pfsense-api#38]
It would appear that regardless of which Boolean param passed for "blockbogons" (true/false) the end result is that blockbogons is set to <null/blank> on API interface create calls. Not setting a value causes PfSense to label the interface as block bogons enabled (within the UI). This creates a bogon firewall rule for the interface as well.

**Request** (api/v1/interface) - POST
```
{
    "client-id": "admin",
    "client-token": "pfsense",
    "if": "vmx2.10",
	"descr": "VLAN10",
	"enable": "true",
	"type": "staticv4",
	"ipaddr": "10.250.0.1",
	"subnet": "24",
	"blockbogons": false
}
```
**Request Response**
```
{
    "status": "ok",
    "code": 200,
    "return": 0,
    "message": "Success",
    "data": {
        "opt2": {
            "if": "vmx2.10",
            "enable": "",
            "descr": "VLAN10",
            "blockbogons": "",
            "ipaddr": "10.250.0.1",
            "subnet": "24"
        }
    }
}
```

As a side note I also noticed that the documented list of params for "/api/v1/interface" does not include a param for **blockbogons**; however, the sample code on github does. Thanks for any insights you might have.

**Comment by riorescue** [comment]
I just noticed that the same behavior occurs for the **enable** param as well. Doesn't matter which Boolean value I send in the request (true/false) quoted or unquoted the end result is the same--the interface is **enabled**.

**Comment by riorescue** [comment]
So this behavior may be expected. The model shows that if either **enable** or **blockbogons** simply exists then set to true.

```
// Check for our enable value
        if (isset($this->initial_data["enable"])) {
            $enable = true;
            $this->validated_data[$next_if]["enable"] = "";
        }

// Check for our block private IP value
        if (isset($this->initial_data["blockbogons"])) {
            $block_bogons = true;
            $this->validated_data[$next_if]["blockbogons"] = "";
        }
```

Removing **blockbogons** from the request does in fact mitigate the issue (blockbogons is not set to true). Removing **enable** from the request payload does set the interface as disabled.

**Comment by jaredhendrickson13** [comment]
Hey!

This is a known issue addressed in #23. It cleans up the interface endpoint a lot and removes most of the sloppy proof-of-concept code I had left over. There is a pre-release build available within that pull request if you want to give it a try. I am probably a couple days or so away from merging those changes. If you do try the pre-release build, note the interface endpoint now contains an `apply` parameter that controls whether the configuration is applied immediately or not. You can read more about it in the documentation page or the readme on that branch. 

Thanks!

**Comment by riorescue** [comment]
Thanks for the update. I looked around prior to submitting but missed the notes for #23 .  Thanks again for the quick response and I really like what you are doing with this project. One day perhaps PfSense will create/finish an API but until then your work goes a long way to making it possible.
